### PR TITLE
AMDGPU/GFX12: Add new dot4 fp8/bf8 instructions

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**
Endoding is VOP3P. Tagged as deep/machine learning instructions. i32 type (v4fp8 or v4bf8 packed in i32) is used for src0 and src1. src0 and src1 have no src_modifiers. src2 is f32 and has src_modifiers: f32 fneg(neg_lo[2]) and f32 fabs(neg_hi[2]).

### CODE REVIEW

File: llvm/include/llvm/IR/IntrinsicsAMDGPU.td [hunk 1]
```diff
@@ -2696,6 +2696,25 @@ def int_amdgcn_udot8 :
...
+    [
+      llvm_i32_ty,   // %a
+      llvm_i32_ty,   // %b
+      llvm_float_ty, // %c
+    ],
+    [IntrNoMem, IntrSpeculatable]
+  >;
+
+def int_amdgcn_fdot4_f32_fp8_bf8 : AMDGPU8bitFloatDot4Intrinsic;
+def int_amdgcn_fdot4_f32_bf8_fp8 : AMDGPU8bitFloatDot4Intrinsic;
```
Comment (arsenm) [inline llvm/include/llvm/IR/IntrinsicsAMDGPU.td:2714]: Why fdot instead of dot?

Comment (mariusz-sikora-at-amd) [reply llvm/include/llvm/IR/IntrinsicsAMDGPU.td]: Done, renamed fdot4 to dot4

Comment (arsenm) [reply llvm/include/llvm/IR/IntrinsicsAMDGPU.td]: I guess this was a mistake previously made with the existing dot intrinsics...

**Review by arsenm — APPROVED** [review]

**Comment by mariusz-sikora-at-amd** [comment]
Rebase to run tests

**Comment by mariusz-sikora-at-amd** [comment]
Rebase to run tests

**Pull request merged**